### ISSUE TITLE
Sending API key to endpoints in request header

### ISSUE THREAD

**Issue by prabhat1790** [springfox/springfox#943]
This seems to be related to https://github.com/springfox/springfox/issues/851
I am unable to send the authorization key in the request header using security context and security schemes, using version 2.2.2. My code looks like this-
@Configuration
@EnableSwagger2
@ComponentScan(basePackages = "com.rokitt.**")
public class SwaggerConfiguration {

    @Bean
    public Docket petApi() {
        return new Docket(DocumentationType.SWAGGER_2).groupName("galactica-api").apiInfo(apiInfo())
                .securitySchemes(Lists.newArrayList(apiKey())).securityContexts(Lists.newArrayList(securityContext())).select()
                .paths(apiPaths()).build();
    }

    private ApiKey apiKey() {
        return new ApiKey("mykey", "api_key", "header");
    }

    private BasicAuth basicAuth() {
        return new BasicAuth("XAuth");
    }

    private Predicate<String> apiPaths() {
        return or(regex("/rest/.*"));
    }

    private Predicate<String> userOnlyEndpoints() {
        return input -> input.contains("user");
    }

    private ApiInfo apiInfo() {
        return new ApiInfoBuilder().title("Galactica APIs").description("Backend APIs for Galactica")
                .termsOfServiceUrl("http://springfox.io").contact("Backend Team").license("Apache License Version 2.0")
                .licenseUrl("https://github.com/springfox/springfox/blob/master/LICENSE").version("2.0").build();
    }

    // these components have been copied from
    // http://springfox.github.io/springfox/docs/current/#getting-started
    // in an attempt to configure authToken in the header while making the api calls
    // but it is still not working
    private SecurityContext securityContext() {
        return SecurityContext.builder().securityReferences(defaultAuth()).forPaths(PathSelectors.regex("/rest/.*")).build();
    }

    private List<SecurityReference> defaultAuth() {
        AuthorizationScope authorizationScope = new AuthorizationScope("global", "accessEverything");
        AuthorizationScope[] authorizationScopes = new AuthorizationScope[1];
        authorizationScopes[0] = authorizationScope;
        return Lists.newArrayList(new SecurityReference("mykey", authorizationScopes));
    }

}

**Comment by prabhat1790** [comment]
Any update?

**Comment by simon-ras** [comment]
I too am having issues with this. It seems as if the swagger-ui provided by springfox is hardcoding the api-key to be a query parameter:

https://github.com/springfox/springfox/blob/master/springfox-swagger-ui/src/web/swagger-ui.html#L56

```javascript
function addApiKeyAuthorization() {
   var key = encodeURIComponent($('#input_apiKey')[0].value);
   if (key && key.trim() != "") {
       var apiKeyAuth = new SwaggerClient.ApiKeyAuthorization("api_key", key, "query");
       window.swaggerUi.api.clientAuthorizations.add("api_key", apiKeyAuth);
       log("added key " + key);
   }
}
```

This should be dynamic somehow

**Comment by dilipkrish** [comment]
@simon-ras thanks for tracking this down... It may have been fixed in the current snapshot

**Comment by prabhat1790** [comment]
Hi Dilip, when is this release going to happen?

**Comment by dilipkrish** [comment]
@prabhat1790 not sure, possibly sometime next week

**Comment by lgringo** [comment]
No, it's not fixed in the current snapshot.
Moreover, it's directly related to swagger-ui. You need to modify index.html of swagger-ui in order to use api_key in headers.

To be precise, there is 2 bugs, index.html does not read the securityConfiguration inside the json :
1.  "query" is hardcoded.
2.  "api_key" is hardcoed.

**Comment by dilipkrish** [comment]
@lgringo Could you confirm if this is fixed or broken in 2.2.3-SNAPSHOT. Planning to release 2.3 shortly and it would be good to close this story if its fixed in the snapshot, before the release.

For one, the index.html is not used in the swagger-ui version thats shipped with this library, we use a custom swagger-ui.html
